lib/galaxy/datatypes/speech.py

```diff
@@ -0,0 +1,120 @@
+from galaxy.datatypes.text import Text
+from galaxy.datatypes.binary import Binary
+from galaxy.datatypes.metadata import MetadataElement, ListParameter, DictParameter
+from galaxy import util
+
+import wave
+import re
+
+import logging
+log = logging.getLogger(__name__)
+
+class WAV( Binary ):
+    """RIFF WAV audio file"""
+
+    file_ext = "wav"
+    blurb = "RIFF WAV Audio file"
+    is_binary = True
+
+    def get_mime(self):
+        """Returns the mime type of the datatype"""
+        return 'audio/wav'
+
+    def sniff(self, filename):
+        """
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname('hello.wav')
+        >>> WAV().sniff(fname)
+        True
+
+        >>> fname = get_test_fname('drugbank_drugs.cml')
+        >>> WAV().sniff(fname)
+        False
+        """
+
+        try:
+            fp = wave.open(filename, 'rb')
+            fp.close()
+            return True
+        except wave.Error:
+            return False
+
+Binary.register_sniffable_binary_format('wav', 'wav', WAV)
+
+class TextGrid( Text ):
+    """Praat Textgrid file for speech annotations
+
+    >>> from galaxy.datatypes.sniff import get_test_fname
+    >>> fname = get_test_fname('1_1119_2_22_001.TextGrid')
+    >>> TextGrid().sniff(fname)
+    True
+
+    >>> fname = get_test_fname('drugbank_drugs.cml')
+    >>> TextGrid().sniff(fname)
+    False
+
+    """
+
+    file_ext = "TextGrid"
+    header = 'File type = "ooTextFile"\nObject class = "TextGrid"\n'
+
+    blurb = "Praat TextGrid file"
+
+    MetadataElement( name="annotations", default=[], desc="Annotation types", param=ListParameter, readonly=True, visible=True, optional=True, no_value=[] )
+
+    def sniff(self, filename):
+
+        with open(filename, 'r') as fd:
+            firstline = fd.readline()
```

Comment:
can you read here just the first N characters, this will avoid reading very very long lines ...

Reply:
good point, will fix that up